### DISCUSSION TITLE
Is it possible to register and unregister APIs (member functions) dynamically at runtime？

### DISCUSSION THREAD

**Discussion by yqs112358** [Tencent/ScriptX#53] [Q&A]
As it said above

**Comment by Jaciezyt** [comment]
@LanderlYoung 大佬打扰了 请问能解答一下吗

我在主引擎dll（负责加载脚本的库）注册原生类时，调用RegScriptClass回调函数传入`ScriptEngine*`指针让其他dll（提供API的库）也注册原生类，但提示不能在`EngineScope`之外调用api。`EngineScope scope(engine)`之后仍发生错误`create EngineScope with a nullptr`

我这个就是在实现动态注册原生类，但很显然失败了

**Reply by LanderlYoung** [reply]
信息有点少，没法判断。但是跨DLL使用C++库本身就需要非常谨慎，尤其是使用了大量模板的场景下。

建议你先合并到一个dll里尝试一下

**Reply by Jaciezyt** [reply]
> 信息有点少，没法判断。但是跨DLL使用C++库本身就需要非常谨慎，尤其是使用了大量模板的场景下。
> 
> 建议你先合并到一个dll里尝试一下

@LanderlYoung 目前就是在一个dll里的，但是编译时间很长，想解耦合
